app/src/controllers/user/actionCreators.js

```diff
@@ -1,11 +1,33 @@
-import { fetch } from 'common/utils';
-import { FETCH_USER_SUCCESS } from './constants';
+import { fetch, getStorageDataByKey } from 'common/utils';
+import { FETCH_USER_SUCCESS, SET_ACTIVE_PROJECT } from './constants';
+import { userInfoSelector } from './selectors';
 
 const fetchUserSuccessAction = user => ({
   type: FETCH_USER_SUCCESS,
   payload: user,
 });
+
+export const setActiveProjectAction = project => (dispatch, getState) => {
+  const user = userInfoSelector(getState());
+  const currentUserSettings = getStorageDataByKey(`${user.userId}_settings`) || {};
+  if (~Object.keys(user.assigned_projects).indexOf(project)) {
```

Comment:
Let's move this check out of this action to the `index.jsx`. Also, we already get user from state there and do some checking, so it will be more consistent to have all checks in one place.

Reply:
fixed